src/main/java/org/jenkinsci/plugins/workflow/log/TaskListenerDecorator.java

```diff
@@ -131,6 +131,14 @@ public interface Factory extends ExtensionPoint {
          */
         @CheckForNull TaskListenerDecorator of(@NonNull FlowExecutionOwner owner);
 
+        /**
+         *
+         * @return boolean, false means to apply step decorators first, then TaskListenerDecorator, true means otherwise
+         * @see #apply(TaskListener, FlowExecutionOwner, TaskListenerDecorator)
+         */
+        default boolean isAppliedBeforeMainDecorator(){
```

Comment:
based off of @dwnusbaum 's [suggestion](https://github.com/jenkinsci/workflow-api-plugin/pull/166), can we add `@Restricted(Beta.class)` to this new method?

Reply:
> based off of @dwnusbaum 's [suggestion](https://github.com/jenkinsci/workflow-api-plugin/pull/166), can we add `@Restricted(Beta.class)` to this new method?

For sure - I thought we had already added that. @pyieh will update this PR with that change hopefully early next week (he's currently enjoying some well deserved time off)